### PR TITLE
Update Pix2Pix for Model Standards

### PR DESCRIPTION
**Pull request by dallasfoster**
# PhysicsNeMo Pull Request

## Description
This PR makes a handful of documentation and argument typing changes in order to better fit model implementation coding standards. We make the conscience choice not to move the resnet block or unet skip connection block because they are mostly model specific and it would require too many changes to upstream those blocks.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [x] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Dependencies

No new dependencies 

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: physicsnemo/models/pix2pix/pix2pix.py [hunk 1]
```diff
@@ -55,9 +55,11 @@
 """
 
 from dataclasses import dataclass
+from typing import Union
 
 import torch
 import torch.nn as nn
+from jaxtyping import Float
```
[the remark this answers is not in the document]
Comment (CharlelieLrt) [reply physicsnemo/models/pix2pix/pix2pix.py:63]: @greptileai you are seriously hallucinating! And who told you to add this rule in your dashboard? Remove this from your dashboard and never do something so dump ever again.

[the remark this answers is not in the document]
Comment (CharlelieLrt) [reply physicsnemo/models/pix2pix/pix2pix.py:63]: @greptileai Okay, you were not too wrong. Actually, jit and jaxtyping are indeed incompatible. But we are in the process of deprecating jit support for most models in the package, so we should instead keep jaxtyping and remove jit. Add the rule to your context and moving forward it's what you will apply for your reviews.

**Comment by coreyjadams** [comment]
I can't remember - was the final decision to deprecate this model instead?

**Comment by CharlelieLrt** [comment]
@coreyjadams yes, no need to enforce any standard and just add a deprecation message. Will definitively deprecate it in next release or so.

**Review by coreyjadams — APPROVED** [review]
Considering this is heading for deprecation, I'll approve as long as we're not breaking CI.

**Comment by coreyjadams** [comment]
FYI we should target RC branch
